### PR TITLE
Fix NewWave+KNNR component

### PR DESCRIPTION
**Pull request by rcannood**
## Describe your changes

* Ensure that NewWave input is a dgCMatrix
* Installing NewWave from GitHub is no longer necessary

## Issue ticket number and link
Closes #xxxx (Replace xxxx with the GitHub issue number)

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [x] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI Tests succeed and look good!

## Requirements after merging

- [x] Need to regenerate `common/` resources: `neurips 2021`

- [x] Need to regenerate task-specific resources. Specify: `predict_modality`

### CODE REVIEW

**Comment by rcannood** [comment]
I agree 7df8cef might solve an issue, but this is a branch for fixing NewWave+KNNR :p

Would you mind reverting 7df8cef and pushing it to a different branch? And do you think we can merge this one?

**Comment by KaiWaldrant** [comment]
I was merging my my fix at the datasetloader side to this branch which involves also fixing the regex otherwise it would not work.

**Review by KaiWaldrant — APPROVED** [review]
LGTM for your part.

File: src/datasets/loaders/openproblems_neurips2021_bmmc/script.py [hunk 1]
```diff
@@ -30,10 +31,20 @@ def remove_mod_prefix(df, mod):
...
+  for key in adata:
+      if isinstance(adata[key], sparse.csr_matrix):
+        adata[key] = sparse.csc_matrix(adata[key])
+  return adata
+      
 
 print("load dataset file", flush=True)
 adata = ad.read_h5ad(par["input"])
 
+# Convert to sparsce csc_matrix
```
Comment (rcannood) [inline src/datasets/loaders/openproblems_neurips2021_bmmc/script.py:44]:
```suggestion
# Convert to sparse csc_matrix
```

File: src/datasets/loaders/openproblems_neurips2021_bmmc/script.py [hunk 2]
```diff
@@ -109,3 +120,7 @@ def remove_mod_prefix(df, mod):
 print("Writing adata to file", flush=True)
 adata_mod1.write_h5ad(par["output_mod1"], compression="gzip")
 adata_mod2.write_h5ad(par["output_mod2"], compression="gzip")
+
+
+
+
```
Comment (rcannood) [inline src/datasets/loaders/openproblems_neurips2021_bmmc/script.py:126]:
```suggestion
```

File: src/datasets/loaders/openproblems_neurips2021_bmmc/script.py [hunk 3]
```diff
@@ -30,10 +31,20 @@ def remove_mod_prefix(df, mod):
   suffix = f"{mod}_"
   df.columns = df.columns.str.removeprefix(suffix)
 
+def convert_matrix(adata):
+  for key in adata:
+      if isinstance(adata[key], sparse.csr_matrix):
+        adata[key] = sparse.csc_matrix(adata[key])
+  return adata
```
Comment (rcannood) [inline src/datasets/loaders/openproblems_neurips2021_bmmc/script.py:38]: inplace is also fine, in this case

```suggestion
```

File: src/datasets/loaders/openproblems_neurips2021_bmmc/script.py [hunk 4]
```diff
@@ -30,10 +31,20 @@ def remove_mod_prefix(df, mod):
...
+      
 
 print("load dataset file", flush=True)
 adata = ad.read_h5ad(par["input"])
 
+# Convert to sparsce csc_matrix
+adata.layers = convert_matrix(adata.layers)
+adata.obsm = convert_matrix(adata.obsm)
+
```
Comment (rcannood) [inline src/datasets/loaders/openproblems_neurips2021_bmmc/script.py:47]: inplace is also fine

```suggestion
convert_matrix(adata.layers)
convert_matrix(adata.obsm)

```

**Pull request merged**